Basic/2DSpaceShooter/Assets/Scripts/Bullet.cs

```diff
@@ -17,15 +19,22 @@ public void Config(ShipControl owner, int damage, bool bounce, float lifetime)
         m_Bounce = bounce;
         if (IsServer)
         {
-            // This is bad code don't use invoke.
-            Invoke(nameof(DestroyBullet), lifetime);
+            StartCoroutine(BulletDestroyCoroutine(lifetime));
         }
     }
 
+    IEnumerator BulletDestroyCoroutine(float lifetime)
+    {
+        yield return new WaitForSeconds(lifetime);
+        DestroyBullet();
+    }
+
     public override void OnNetworkDespawn()
     {
-        // This is inefficient, the explosion object could be pooled.
-        GameObject ex = Instantiate(explosionParticle, transform.position + new Vector3(0, 0, -2), Quaternion.identity);
+        m_ObjectPool = GameObject.FindWithTag(s_ObjectPoolTag).GetComponent<NetworkObjectPool>();
```

Comment:
`FindWithTag` is even worse (performance-wise) than a single instantiate, as it searches the entire scene and 

Can you please revert this change?

If you want to use a pool, it should be statically accessible (better) / you should have a direct reference to it (worse)

Reply:
I will revert the change. But I would love to knowhow I can make a pool statically accessible?

Reply:
You'll need to store a static reference to the NetworkObjectPool somewhere. [Looking at the documentation](https://docs-multiplayer.unity3d.com/netcode/current/advanced-topics/object-pooling/), it seems it's already done by the NetworkObjectPool class itself ( `NetworkObjectPool Singleton` )